rust/core-lib/src/editor.rs

```diff
@@ -767,6 +806,22 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.view.set_dirty();
     }
 
+    fn debug_test_soft_spans(&mut self) {
+        print_err!("adding soft spans example");
+
+        let sel_interval = Interval::new_closed_open(self.view.sel_start, self.view.sel_end);
+        let text = Rope::from("soft spans!");
+        let new_cursor = self.view.sel_start;
+
+        let end = new_cursor + text.len();
+        let mut sb = SpansBuilder::new(end);
+        sb.add_span(Interval::new_closed_open(new_cursor, new_cursor + 3), ());
+        sb.add_span(Interval::new_closed_open(new_cursor + 3, end), ());
+        self.soft_spans = sb.build();
+
+        self.add_delta(sel_interval, text, new_cursor, new_cursor);
```

Comment:
I might do something wrong here, because undo does not work properly with the text inserted by this method